sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/text/TextScanBuilder.scala

```diff
@@ -33,6 +33,6 @@ case class TextScanBuilder(
   extends FileScanBuilder(sparkSession, fileIndex, dataSchema) {
 
   override def build(): Scan = {
-    TextScan(sparkSession, fileIndex, readDataSchema(), readPartitionSchema(), options)
+    TextScan(sparkSession, fileIndex, schema, readDataSchema(), readPartitionSchema(), options)
```

Comment:
shall we pass `dataSchema`?

Reply:
Thanks, fixed in https://github.com/apache/spark/pull/33825/commits/7c606943e81888917988dd252719f06052fc6f56.